### PR TITLE
Support Python 3.8, 3.9 and 3.10, drop EOL Python versions (2.7, 3.4, 3.5), use GitHub Actions for CI

### LINKED ISSUE #219: Support virtualenv 20+
The virtualenv project had undergone [a full rewrite](https://discuss.python.org/t/virtualenv-20-0-0-beta1-is-available/3077) in the beginning of 2020 and Pew would need to be updated for that.

Here is a summary of the test failures with virtualenv 20.0.28 on Python 3.9.0 beta 5:
```
___________________________ test_source_relocatable ____________________________
workon_home = PosixPath('/tmp/WORKON_HOME'), testpackageenv = None
    def test_source_relocatable(workon_home, testpackageenv):
>       check_call([executable, '-m', 'virtualenv', '--relocatable',
                    str(workon_home / 'source')])
tests/test_cp.py:40: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
popenargs = (['/usr/bin/python3', '-m', 'virtualenv', '--relocatable', '/tmp/WORKON_HOME/source'],)
kwargs = {}, retcode = 2
cmd = ['/usr/bin/python3', '-m', 'virtualenv', '--relocatable', '/tmp/WORKON_HOME/source']
    def check_call(*popenargs, **kwargs):
        """Run command with arguments.  Wait for command to complete.  If
        the exit code was zero then return, otherwise raise
        CalledProcessError.  The CalledProcessError object will have the
        return code in the returncode attribute.
    
        The arguments are the same as for the call function.  Example:
    
        check_call(["ls", "-l"])
        """
        retcode = call(*popenargs, **kwargs)
        if retcode:
            cmd = kwargs.get("args")
            if cmd is None:
                cmd = popenargs[0]
>           raise CalledProcessError(retcode, cmd)
E           subprocess.CalledProcessError: Command '['/usr/bin/python3', '-m', 'virtualenv', '--relocatable', '/tmp/WORKON_HOME/source']' returned non-zero exit status 2.
/usr/lib64/python3.9/subprocess.py:373: CalledProcessError
----------------------------- Captured stdout call -----------------------------
SystemExit: 2
----------------------------- Captured stderr call -----------------------------
usage: virtualenv [--version] [--with-traceback] [-v | -q] [--app-data APP_DATA] [--reset-app-data] [--upgrade-embed-wheels] [--discovery {builtin}] [-p py] [--creator {builtin,cpython3-posix,venv}] [--seeder {app-data,pip}] [--no-seed]
                  [--activators comma_sep_list] [--clear] [--system-site-packages] [--symlinks | --copies] [--no-download | --download] [--extra-search-dir d [d ...]] [--pip version] [--setuptools version] [--wheel version] [--no-pip]
                  [--no-setuptools] [--no-wheel] [--no-periodic-update] [--symlink-app-data] [--prompt prompt] [-h]
                  dest
virtualenv: error: unrecognized arguments: --relocatable
_____________________________ test_lssitepackages ______________________________
workon_home = PosixPath('/tmp/WORKON_HOME')
    def test_lssitepackages(workon_home):
        invoke('new', 'env', '-d')
        pkgs = invoke('in', 'env', 'pew', 'lssitepackages').out
>       assert 'easy_install' in pkgs
E       AssertionError: assert 'easy_install' in '.git\n.gitignore\n.travis.yml\nLICENSE\nMANIFEST.in\nPipfile\nREADME.Fedora.md\nREADME.md\nappveyor.yml\nbuild\ndebia...debugsources.list\ndefault.nix\nelfbins.list\npew\npew.egg-info\nrequirements.txt\nsetup.cfg\nsetup.py\ntests\ntox.ini'
tests/test_ls.py:20: AssertionError
____________________________ test_create_in_symlink ____________________________
workon_sym_home = PosixPath('/tmp/pew-test-tmp/SYM_ROOT/WORKON_HOME')
    @skip_windows(reason="symlinks on windows are not well supported")
    def test_create_in_symlink(workon_sym_home):
        invoke('new', 'env', '-d')
        pip_path = Path(invoke('in', 'env', 'which', 'pip').out)
        with pip_path.open() as f:
            pip_shebang = f.readline()
        # check it is using the symlinked path, not the real path
>       assert str(workon_sym_home) in pip_shebang
E       AssertionError: assert '/tmp/pew-test-tmp/SYM_ROOT/WORKON_HOME' in '#!/tmp/pew-test-tmp/REAL_ROOT/WORKON_HOME/env/bin/python\n'
E        +  where '/tmp/pew-test-tmp/SYM_ROOT/WORKON_HOME' = str(PosixPath('/tmp/pew-test-tmp/SYM_ROOT/WORKON_HOME'))
tests/test_mkvirtualenv.py:27: AssertionError
_________________________________ test_restore _________________________________
workon_home = PosixPath('/tmp/WORKON_HOME'), env1 = None
    def test_restore(workon_home, env1):
        patterns = ['lib*/*/site.py*', 'Lib/site.py*']
        to_be_deleted = set(chain(*((workon_home / 'env1').glob(pat) for pat in patterns)))
        for site in to_be_deleted:
            try:
                site.unlink()
            except FileNotFoundError:
                pass # multiple links to the same file might appear in to_be_deleted
        result = invoke('in', 'env1', 'python', '-vc', '')
>       assert 'Error' in result.err or 'fail' in result.err
E       assert ('Error' in "import _frozen_importlib # frozen\nimport _imp # builtin\nimport '_thread' # <class '_frozen_importlib.BuiltinImporte...troy io\n# destroy _warnings\n# destroy _thread\n# destroy posix\n# destroy _frozen_importlib\n# clear sys.audit hooks" or 'fail' in "import _frozen_importlib # frozen\nimport _imp # builtin\nimport '_thread' # <class '_frozen_importlib.BuiltinImporte...troy io\n# destroy _warnings\n# destroy _thread\n# destroy posix\n# destroy _frozen_importlib\n# clear sys.audit hooks")
E        +  where "import _frozen_importlib # frozen\nimport _imp # builtin\nimport '_thread' # <class '_frozen_importlib.BuiltinImporte...troy io\n# destroy _warnings\n# destroy _thread\n# destroy posix\n# destroy _frozen_importlib\n# clear sys.audit hooks" = Result(returncode=0, out='', err='import _frozen_importlib # frozen\nimport _imp # builtin\nimport \'_thread\' # <clas...roy io\n# destroy _warnings\n# destroy _thread\n# destroy posix\n# destroy _frozen_importlib\n# clear sys.audit hooks').err
E        +  and   "import _frozen_importlib # frozen\nimport _imp # builtin\nimport '_thread' # <class '_frozen_importlib.BuiltinImporte...troy io\n# destroy _warnings\n# destroy _thread\n# destroy posix\n# destroy _frozen_importlib\n# clear sys.audit hooks" = Result(returncode=0, out='', err='import _frozen_importlib # frozen\nimport _imp # builtin\nimport \'_thread\' # <clas...roy io\n# destroy _warnings\n# destroy _thread\n# destroy posix\n# destroy _frozen_importlib\n# clear sys.audit hooks').err
tests/test_restore.py:15: AssertionError
```

The upcoming Fedora 33 will feature both virtualenv 20 and Python 3.9. Here is the [downstream bug report about Pew failing to build from source](https://bugzilla.redhat.com/show_bug.cgi?id=1865212).

### PR DESCRIPTION
**Pull request by tjanez**
This PR:

- adds support for Python 3.8, 3.9 and 3.10,
- drops support for EOL Python versions (2.7, 3.4, 3.5),
- replaces Travis CI and AppVeyor with GitHub Actions,
- updates/fixes tests for new virtualenv (version 20+).

TODO:

- [x] fix `test_lssitepackages` test failure
- [x] fix `test_restore` test failure
- [x] temporarily skip the `test_create_in_symlink` test due to strange failure (see: #225)

Fixes #219.

~This will start testing Pew with ~the upcoming~ Python 3.8.~

~_NOTE: The Travis tests are configured to use the "3.8-dev" version and should be switched to "3.8" once it is released._~

~I've also pinned virtualenv dependency to `<20.0.0` (support for virtualenv 20+ is tracked in #219) for now and rewritten the AppVeyor configuration to fix test issues there.~

### CODE REVIEW

**Comment by tjanez** [comment]
@pfmoore, this is ready for review.

**Comment by pfmoore** [comment]
Travis build is failing. Also, needs confirmation that dropping support for the older Python versions is acceptable (I'm OK with doing so myself, but it's not my call).

**Comment by tjanez** [comment]
> Travis build is failing.

Yes, the Nix job is failing but apparently, [it has been failing for a while](https://travis-ci.org/github/berdario/pew/builds) since the last successful build on the `master` branch is from Oct 5, 2019 ([Build 557](https://travis-ci.org/github/berdario/pew/builds/593777224)).

> Also, needs confirmation that dropping support for the older Python versions is acceptable (I'm OK with doing so myself, but it's not my call).

Ok, let's wait for @berdario.

**Comment by tjanez** [comment]
> Travis build is failing. Also, needs confirmation that dropping support for the older Python versions is acceptable (I'm OK with doing so myself, but it's not my call).

@pfmoore, I've removed the dropping of Python 3.4 commit to be able to get this PR merged quicker.

> > Travis build is failing.
> 
> Yes, the Nix job is failing but apparently, [it has been failing for a while](https://travis-ci.org/github/berdario/pew/builds) since the last successful build on the `master` branch is from Oct 5, 2019 ([Build 557](https://travis-ci.org/github/berdario/pew/builds/593777224)).
> 

I've filled #221 and temporarily allowed Nix build jobs to fail on Travis CI since this issue is unrelated to this (or other open) PRs.

Please, take another look at this PR.

File: setup.py [hunk 1]
```diff
@@ -62,21 +62,12 @@ def run(self):
     license='MIT License',
     packages=['pew'],
     install_requires=[
-        'virtualenv>=1.11', 'virtualenv-clone>=0.2.5', 'setuptools>=17.1'
+        'virtualenv<20.0.0,>=1.11', 'virtualenv-clone>=0.2.5', 'setuptools>=17.1'
```
Comment (pfmoore) [inline setup.py:65]: I use virtualenv 20.0 with pew regularly. It works fine for the features I use, and is *far* faster than the older version. So I'm a strong -1 on forcing an older version here.

If there are pew features that are not available with the new virtualenv version, they should fail individually.

**Review by pfmoore — DISMISSED** [review]

**Comment by pfmoore** [comment]
Still needs @berdario to comment. But I have pointed out that I don't want the project to force an older version of virtualenv. So I'd rather the virtualenv wasn't pinned, but #219 gets addressed. I'd object to a release that pinned to the old version, so I see no point in adding a pin now, just to have taking it out again be a release blocker...

**Pull request merged**